src/base/action.lua

```diff
@@ -257,6 +257,47 @@
 		return false
 	end
 
+---
+-- Determines if an action supports a particular toolset.
+--
+-- @param language
+--    The language that toolset belongs.
+-- @param toolset
+--    The toolset to check.
+-- @returns
+--    True if the toolset is supported, false otherwise.
+---
+
+	function action.supportsToolset(language, toolset)
+		local function buildLanguageKeysMap()
+			local language_keys_map = {C = "cc", D = "dc" }
+			language_keys_map["C++"] = "cc"
+			language_keys_map["C#"] = "dotnet"
+			return language_keys_map
+		end
+		if not language or not toolset then
+			return true
+		end
+		local self = action.current()
+		if not self then
+			return false
+		end
+		local language_keys_map = buildLanguageKeysMap()
+		local language_key = language_keys_map[language]
+		if not language_key then
+			print("unknown mapping for language %s", language)
```

Comment:
Was the choice of `print` based on another similar function, or should this be `p.warn`?

Reply:
Changed to `p.warn`